dll/cpl/hotplug/hotplug.c

```diff
@@ -344,6 +344,78 @@ ShowContextMenu(
 }
 
 
+static
+DWORD
+GetSelectedDeviceInst(
+    _In_ HWND hwndDeviceTree)
+{
+    HTREEITEM hTreeItem;
+    TVITEMW item;
+
+    hTreeItem = TreeView_GetSelection(hwndDeviceTree);
+    if (hTreeItem == NULL)
+        return 0;
+
+    ZeroMemory(&item, sizeof(item));
+    item.mask = TVIF_PARAM;
+    item.hItem = hTreeItem;
+
+    TreeView_GetItem(hwndDeviceTree, &item);
+
+    return item.lParam;
+}
+
+static
+int
+_DevicePropertiesW(
+    _In_ HWND hWndParent,
+    _In_ LPCWSTR lpMachineName,
+    _In_ LPCWSTR lpDeviceID,
```

Comment:
```suggestion
    _In_opt_ HWND hWndParent,
    _In_opt_ LPCWSTR lpMachineName,
    _In_opt_ LPCWSTR lpDeviceID,
```
In principle all these parameters are optional (by checking the implementation of DeviceProperties(Ex)W)

Reply:
In addition, I don't know whether you plan to use this DevicePropertiesW with other sorts of parameters later on, but if not, you may instead consider simplifying its implementation, much like what is done here for example:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/shellext/deskadp/deskadp.c;hb=855008d97b2c7e1a1b72ad268b68d83c739ebc6d#l267

Reply:
Function removed.